agimus_controller_ros/agimus_controller_ros/simple_trajectory_publisher.py

```diff
@@ -90,12 +92,9 @@ def publish_mpc_input(self):
         Modifies each joint in sin manner with 0.2 rad amplitude
         """
 
-        if self.pin_model is None:  # wait for model to be available
-            return
-        if self.q0 is None or self.q is None:
-            return
-        if self.delay_time > 0:
-            self.delay_time -= 1
+        if self.robot_description_msg is not None and self.q0 is not None:
+            self.load_models()
```

Comment:
to avoid calling this even when the models are loaded I would do 
```suggestion
        if self.robot_description_msg is not None and self.q0 is not None and self.pin_model is None:
            self.load_models()
```